### ISSUE TITLE
lightning stops building

### ISSUE THREAD

**Issue by renepickhardt** [ElementsProject/lightning#2879]
I don't know why but our current master stopped building even in a clean environment and after running `make clean` the issue seems to be `tool/mockup.sh: line 11:`		;;`

I have no clue at all why it stopped working and I am not that deep into the build process of c code.

so even in a completely fresh git clone I run into this issue after `./configure` and `make`

operating system is mac-os. I can later check on a linux machine but I guess if the problem was there too you would already know.

```
tools/generate-wire.py --page impl tools/test/gen_test.h test_type < tools/test/test_cases > 
tools/test/gen_test.c:
tools/mockup.sh: line 11: syntax error near unexpected token `;;'
tools/mockup.sh: line 11: `		;;'
make: *** [tools/test/gen_test.c] Error 2
rm external/libwally-core/src/secp256k1/libsecp256k1.la
Renes-iMacReal:lightning rpickhardt$ 
``` 

## git log
```
commit 0ae20399bd05adf8b6c1827359d9ae42e5f7376b (HEAD -> master, origin/master, origin/HEAD)
Author: trueptolemy <823220586@qq.com>
Date:   Sun Jul 28 21:24:57 2019 +0800

    CHANGELOG: check the bitcoind version when `setup_topology`

commit 130bf20516be815acc5034ba84b9c106ac6274a9
Author: trueptolemy <823220586@qq.com>
Date:   Sun Jul 28 01:58:27 2019 +0800

    lightningd: check bitcoind version when `setup_topology`
```

**Comment by renepickhardt** [comment]
actually on linux I have 

```
tools/generate-wire.py --page header common/gen_status_wire.h status < common/status_wire.csv > common/gen_status_wire.h
Traceback (most recent call last):
  File "tools/generate-wire.py", line 24, in <module>
    from mako.template import Template
ModuleNotFoundError: No module named 'mako'
common/Makefile:83: recipe for target 'common/gen_status_wire.h' failed
make: *** [common/gen_status_wire.h] Error 1
rm external/libwally-core/src/secp256k1/libsecp256k1.la
```
indicating that a pythong dependency for the tests is missing. was able to fix this with
`pip install mako` will open a PR in 1 minute in which I add the mako dependency to  `tests/requirements.txt`

**Comment by renepickhardt** [comment]
other than that on ubuntu lighting builds so I guess it is a mac issue and thus probably harder to fix

**Comment by ZmnSCPxj** [comment]
Indeed, looks like a Mac issue.  Maybe Mac is using a faked `bash`.  Can you try:

    ls -la `which bash`

**Comment by rustyrussell** [comment]
Mako is now a global dependency; it's in doc/INSTALL.md.  If you didn't have it, you might have got empty files: try make distclean?

Putting it in requirements.txt for tests/ is a bit misplaced, but can't hurt.

**Comment by renepickhardt** [comment]
> Indeed, looks like a Mac issue. Maybe Mac is using a faked `bash`. Can you try:
> 
> ```
> ls -la `which bash`
> ```

Renes-iMacReal:regtest-helper rpickhardt$ ls -la `which bash`
-r-xr-xr-x  1 root  wheel  618448 21 Sep  2017 /bin/bash

**Comment by renepickhardt** [comment]
> Mako is now a global dependency; it's in doc/INSTALL.md. If you didn't have it, you might have got empty files: try make distclean?

I checkted that file right now and I only found a reference in the ubuntu build instructions but not for mac-os and in the beginning with the global dependencies it was also not mentioned.

**Comment by Sjors** [comment]
I also ran into the missing mako issue when trying to make 0.7.2. Probably good to mention in the release notes.

**Comment by gabridome** [comment]
I run the same in
`Linux 4.9.0-9-amd64 #1 SMP Debian 4.9.168-1+deb9u2 (2019-05-13) x86_64 GNU/Linux`

`make install` doesn't solve.

**Comment by rustyrussell** [comment]
AI cleanup of old issues.  If wrong, please open a new issue.


Triage decision summary:
Build failure on macOS in tools/mockup.sh with shell syntax error. Issue is from July 2019 with no recent activity or follow-up, suggesting it was either fixed in subsequent commits or the build system was refactored.
